<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Methyl-Seq Workflow Template • systemPipeMethylseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Methyl-Seq Workflow Template">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158796879-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158796879-1');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeMethylseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a></a>
</li>
<li>
  <a href="../articles/systemPipeMethylseq.html">Get started</a>
</li>
<li>
  <a href="../articles/systemPipeMethylseq.html">
    <span class="fas fa fas fa-book"></span>
     
    Workflow
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tgirke/systemPipeR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Methyl-Seq Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 22 February, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/systemPipeMethylseq/blob/master/vignettes/systemPipeMethylseq.Rmd"><code>vignettes/systemPipeMethylseq.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeMethylseq.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!--
## Compile from command-line
Rscript -e "rmarkdown::render('systemPipeMethylseq.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeMethylseq.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeMethylseq.knit.md 2; Rscript -e "rmarkdown::render('systemPipeMethylseq.Rmd', c('BiocStyle::pdf_document'))" --><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Users want to provide here background information about the design of their Methyl-Seq project.</p>
<p>Cytosine methylation is one of the important epigenetic mechanism to control gene expression. The epigenetic modification is has been associated with many biological processes, including various diseases and normal development, as well. One of the techniques for profiling genome-wide DNA methylation is based on sodium bisulfite, followed by NGS (BS-seq). Whole-genome bisulfite sequencing (WGBS) provides global coverage at a single-base resolution. This protocol effectively converts on-methylated cytosines into uracils, while methylated cytosines are preserved. Eventually, uracils are converts into thymines during the PCR amplification cycles.</p>
<!-- Add some introduction about the sequencing protocol... -->
</div>
<div id="sample-definitions-and-environment-settings" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-definitions-and-environment-settings" class="anchor"></a>Sample definitions and environment settings</h1>
<div id="environment-settings-and-input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#environment-settings-and-input-data" class="anchor"></a>Environment settings and input data</h2>
<p>Typically, the user wants to record here the sources and versions of the reference genome sequence along with the corresponding annotations. In the provided sample data set all data inputs are stored in a <code>data</code> subdirectory and all results will be written to a separate <code>results</code> directory, while the <code>systemPipeMethylseq.Rmd</code> script and the <code>targets</code> file are expected to be located in the parent directory. The R session is expected to run from this parent directory.</p>
<p>The chosen data set <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRP097911">SRP097911</a> contains single-end (SE) read sets from <em>Arabidposis thaliana</em> <span class="citation">(Ganguly et al. 2017)</span>.</p>
<!--
TODO: remover this part and add the new explain form the new genWorkenvir function)
-->
<p>The following loads one of the available NGS workflow templates (here Methyl-Seq) into the user’s current working directory. At the moment, the package includes workflow templates for RNA-Seq, ChIP-Seq, VAR-Seq, Ribo-Seq and Methyl-Seq. Templates for additional NGS applications will be provided in the future.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeRdata)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeRdata/man/genWorkenvir.html">genWorkenvir</a></span>(<span class="dt">workflow =</span> <span class="ot">NULL</span>, <span class="dt">package_repo =</span> <span class="st">"systemPipeR/systemPipeMethylseq"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span>(<span class="st">"systemPipeMethylseq"</span>)</a></code></pre></div>
<p>Now open the R markdown script <code>systemPipeMethylseq.Rmd</code>in your R IDE (<em>e.g.</em> vim-r or RStudio) and run the workflow as outlined below.</p>
</div>
<div id="required-packages-and-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#required-packages-and-resources" class="anchor"></a>Required packages and resources</h2>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(Girke 2014)</span>. <!--
TODO: add new package dependence in the Description, and namespace
--></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeR)</a></code></pre></div>
</div>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">targetspath &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment.char =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">targets[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</a></code></pre></div>
<pre><code>##                      FileName SampleName Factor SampleLong
## 1 ./data/SRR446027_1.fastq.gz        M1A     M1  Mock.1h.A
## 2 ./data/SRR446028_1.fastq.gz        M1B     M1  Mock.1h.B
## 3 ./data/SRR446029_1.fastq.gz        A1A     A1   Avr.1h.A
## 4 ./data/SRR446030_1.fastq.gz        A1B     A1   Avr.1h.B</code></pre>
</div>
</div>
<div id="read-preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#read-preprocessing" class="anchor"></a>Read preprocessing</h1>
<div id="read-quality-filtering-and-trimming" class="section level2">
<h2 class="hasAnchor">
<a href="#read-quality-filtering-and-trimming" class="anchor"></a>Read quality filtering and trimming</h2>
<p>The function <code>preprocessReads</code> allows to apply predefined or custom read preprocessing functions to all FASTQ files referenced in a <code>SYSargs2</code> container, such as quality filtering or adapter trimming routines. The paths to the resulting output FASTQ files are stored in the <code>output</code> slot of the <code>SYSargs2</code> object. The following example performs adapter trimming with the <code>trimLRPatterns</code> function from the <code>Biostrings</code> package. After the trimming step a new targets file is generated (here <code>targets_trim.txt</code>) containing the paths to the trimmed FASTQ files. The new targets file can be used for the next workflow step with an updated <code>SYSargs2</code> instance, <em>e.g.</em> running the NGS alignments using the trimmed FASTQ files.</p>
<p>Construct <em><code>SYSargs2</code></em> object from <em><code>cwl</code></em> and <em><code>yml</code></em> param and <em><code>targets</code></em> files.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/preprocessReads/trim-se"</span>, </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">    <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="dt">targets =</span> targetspath, <span class="dt">wf_file =</span> <span class="st">"trim-se.cwl"</span>, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">    <span class="dt">input_file =</span> <span class="st">"trim-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(trim, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">trim</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">output</a></span>(trim)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/preprocessReads.html">preprocessReads</a></span>(<span class="dt">args =</span> trim, <span class="dt">Fct =</span> <span class="st">"trimLRPatterns(Rpattern='GCCCGGGTAA', </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="st">                subject=fq)"</span>, </a>
<a class="sourceLine" id="cb6-3" data-line-number="3">    <span class="dt">batchsize =</span> <span class="fl">1e+05</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>, <span class="dt">compress =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> trim, <span class="dt">file =</span> <span class="st">"targets_trim.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb6-5" data-line-number="5">    <span class="dt">new_col =</span> <span class="st">"FileName1"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="fastq-quality-report" class="section level2">
<h2 class="hasAnchor">
<a href="#fastq-quality-report" class="anchor"></a>FASTQ quality report</h2>
<p>The following <code>seeFastq</code> and <code>seeFastqPlot</code> functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box plots, base proportions, base-level quality trends, relative k-mer diversity, length and occurrence distribution of reads, number of reads above quality cutoffs and mean quality distribution. The results are written to a PDF file named <code>fastqReport.pdf</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">fqlist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastq</a></span>(<span class="dt">fastq =</span> <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(trim), <span class="dt">batchsize =</span> <span class="dv">10000</span>, </a>
<a class="sourceLine" id="cb7-2" data-line-number="2">    <span class="dt">klength =</span> <span class="dv">8</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/fastqReport.pdf"</span>, <span class="dt">height =</span> <span class="dv">18</span>, <span class="dt">width =</span> <span class="dv">4</span> <span class="op">*</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(fqlist))</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastqPlot</a></span>(fqlist)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<p><img src="results/fastqReport.png"></p>
<div align="center">
Figure 1: FASTQ quality report for 6 samples before filtering.
</div>
<p></p>
</div>
</div>
<div id="alignments" class="section level1">
<h1 class="hasAnchor">
<a href="#alignments" class="anchor"></a>Alignments</h1>
<div id="read-mapping-with-bowtiebismark" class="section level2">
<h2 class="hasAnchor">
<a href="#read-mapping-with-bowtiebismark" class="anchor"></a>Read mapping with <code>Bowtie/Bismark</code>
</h2>
<p>The NGS reads of this project will be aligned against the reference genome sequence using <code>Bowtie/Bismark</code> <span class="citation">(Langmead and Salzberg 2012, <span class="citation">@Krueger2011-uu</span>)</span> in both interactive job submissions and batch submissions to queuing systems of clusters using the <em><code>systemPipeR's</code></em> new CWL command-line interface.</p>
<!-- 
Get better explanation
-->
<p>Bisulfite sequencing aligners, like Bismark <span class="citation">Krueger and Andrews (2011)</span>] are based on three-letter algorithm. Which converts all Cs in the reference genome and the read into Ts, and thus, standard aligners with lower mappability can be adopted due to reduced sequence complexity.</p>
<p>Build <em><code>Bismark</code></em> index.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">dir_path &lt;-<span class="st"> "param/cwl/bismark/bismark_idx/"</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">idx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="dt">targets =</span> <span class="ot">NULL</span>, <span class="dt">wf_file =</span> <span class="st">"bismark-index.cwl"</span>, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"bismark-index.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">idx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(idx)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">idx</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">cmdlist</a></span>(idx)</a>
<a class="sourceLine" id="cb8-7" data-line-number="7"></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">## Run in single machine</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(idx, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>The parameter settings of the aligner are defined in the <code>bismark-mapping.cwl</code> and <code>bismark-mapping.yml</code> files. The following shows how to construct the corresponding <em>SYSargs2</em> object, here <em>args</em>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">dir_path &lt;-<span class="st"> "param/cwl/bismark/bismark/"</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="dt">targets =</span> targetspath, <span class="dt">wf_file =</span> <span class="st">"bismark-mapping-se.cwl"</span>, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"bismark-mapping-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb9-5" data-line-number="5">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">args</a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">cmdlist</a></span>(args)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">output</a></span>(args)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">## Run in single machine</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>Check whether all BAM files have been created and write the new <code>targets file</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(outpaths)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"R_function/writeTargetsout2.R"</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="kw">writeTargetsout2</span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_bismark.txt"</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb10-5" data-line-number="5">    <span class="dt">folder =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="read-and-plot-the-alignment-stats" class="section level2">
<h2 class="hasAnchor">
<a href="#read-and-plot-the-alignment-stats" class="anchor"></a>Read and plot the alignment stats</h2>
<p>The following step provides an overview of the number of reads in each sample and how many of them aligned to the reference. By default, in the alignment process, after the conversion to bedGraph has completed, the option –cytosine_report produces a genome-wide methylation report for all cytosines in the genome. The following step generates a graphics output with methylation levels.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"R_function/read.align.bs.R"</span>)  <span class="co">## FIX rownames TABLE.</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">align_stats &lt;-<span class="st"> </span><span class="kw">read.align.bs</span>()</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">align_stats[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<img src="results/align.bs_plot.png"><div align="center">
Figure 3: Alignment status.
</div>
<p></p>
<img src="results/methyl.levels_plot.png"><div align="center">
Figure 4: Methylation Levels across the samples.
</div>
<p></p>
</div>
<div id="methylation-calling" class="section level2">
<h2 class="hasAnchor">
<a href="#methylation-calling" class="anchor"></a>Methylation Calling</h2>
<p>In order to extract the methylation call for every single C analysed from the Bismark alignment files, we used the <code>bismark_methylation_extractor</code> function. The parameter settings of the function are defined in the <code>bismark_methyl_extractor.param</code> file. Submission of Bismark_extract as a single process.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/bismark_methyl_extractor.param"</span>, </a>
<a class="sourceLine" id="cb12-2" data-line-number="2">    <span class="dt">mytargets =</span> <span class="st">"targets_bismark.txt"</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span>(args)[<span class="dv">1</span>]  <span class="co"># Command-line parameters for first FASTQ file </span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"runCommandline.bs.R"</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span>(args))) {</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">    <span class="kw"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span>(args)[i], <span class="dt">showWarnings =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span>(args))</a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="kw">runCommandline_bs</span>(args)</a></code></pre></div>
<p><img src="results/SRR5494752.fastq_trim.gz_bismark_bt2.M-bias_R1.png"></p>
<div align="center">
Figure 5: Methylation proportion acrros each possible position in the read, in the first sample.
</div>
<p></p>
<p>Alternatively, submit jobs to compute cluster.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">resources &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">walltime =</span> <span class="dv">1440</span>, <span class="dt">ntasks =</span> <span class="dv">1</span>, <span class="dt">ncpus =</span> <span class="dv">14</span>, <span class="dt">memory =</span> <span class="dv">10240</span>, </a>
<a class="sourceLine" id="cb13-2" data-line-number="2">    <span class="dt">email =</span> <span class="st">"danielac@ucr.edu"</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">reg &lt;-<span class="st"> </span><span class="kw">clusterRun2</span>(args, <span class="dt">FUN =</span> runCommandline_bs, <span class="dt">conf.file =</span> <span class="st">".batchtools.conf.R"</span>, </a>
<a class="sourceLine" id="cb13-4" data-line-number="4">    <span class="dt">template =</span> <span class="st">"batchtools.slurm.tmpl"</span>, <span class="dt">runid =</span> <span class="st">"01"</span>, <span class="dt">resourceList =</span> resources)</a></code></pre></div>
</div>
</div>
<div id="descriptive-statistics-on-dna-methylation-profiles" class="section level1">
<h1 class="hasAnchor">
<a href="#descriptive-statistics-on-dna-methylation-profiles" class="anchor"></a>Descriptive statistics on DNA methylation profiles</h1>
<p>The coverage per base and Percentage of methylation per base are calculeted using the <code>methylKit</code> R package [Akalin2012-ez].</p>
<div id="import-of-bismark-methylation-output-files" class="section level2">
<h2 class="hasAnchor">
<a href="#import-of-bismark-methylation-output-files" class="anchor"></a>Import of Bismark methylation output files</h2>
<p>The <code>Bisrmark</code> results can be loaded with <code>methRead</code> function. Note that a list containing the paths of the cytosine report files from Bismark needs to be provided, which contains chromosome, position, strand, count methylated, count unmethylated, C-context and trinucleotide context format (output file <code>*.cov</code>. In this case, use <code>pipeline='bismarkCoverage</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(methylKit)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"R_function/run_methylKit.R"</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">targetspath &lt;-<span class="st"> "targets.txt"</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">cmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span>(<span class="dt">file =</span> targetspath, <span class="dt">format =</span> <span class="st">"matrix"</span>, <span class="dt">delim =</span> <span class="st">"-"</span>)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/bismark_methyl_extractor.param"</span>, </a>
<a class="sourceLine" id="cb14-7" data-line-number="7">    <span class="dt">mytargets =</span> <span class="st">"targets_bismark.txt"</span>)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">import.bs &lt;-<span class="st"> </span><span class="kw">import.bs.data</span>(targets, args, <span class="dt">cmp =</span> cmp[[<span class="dv">1</span>]])</a></code></pre></div>
<p>Alternatively, methylation results can be calculated from sorted SAM or BAM files from Bismark aligner. The coordinate sorted so an extra re-sort step is required, and <code>Samtools</code>[Li2011-qu] are used. The parameter settings of <code>Samtools</code> are defined in the <code>samtools.param</code> file.</p>
<p>The following provides the code on a single machine, but the processing can be submitted to a compute cluster, as well.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co">## Please note this step may take a while ...</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"clusterRun2.R"</span>)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"runCommandline2.R"</span>)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/samtools.param"</span>, <span class="dt">mytargets =</span> <span class="st">"targets_bismark.txt"</span>)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">sysargs</a></span>(args)[<span class="dv">1</span>]  <span class="co"># Command-line parameters for first FASTQ file </span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb15-7" data-line-number="7"></a>
<a class="sourceLine" id="cb15-8" data-line-number="8">report1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outfile1</a></span>(args))</a>
<a class="sourceLine" id="cb15-9" data-line-number="9">sample.id &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(targets<span class="op">$</span>SampleName)</a>
<a class="sourceLine" id="cb15-10" data-line-number="10">methRaw &lt;-<span class="st"> </span><span class="kw">processBismarkAln</span>(<span class="dt">location =</span> report1, <span class="dt">sample.id =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(sample.id), </a>
<a class="sourceLine" id="cb15-11" data-line-number="11">    <span class="dt">assembly =</span> <span class="st">"TAIR10"</span>, <span class="dt">save.folder =</span> <span class="st">"results/"</span>, <span class="dt">save.context =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb15-12" data-line-number="12">    <span class="dt">read.context =</span> <span class="st">"CpG"</span>, <span class="dt">nolap =</span> <span class="ot">FALSE</span>, <span class="dt">mincov =</span> <span class="dv">10</span>, <span class="dt">minqual =</span> <span class="dv">20</span>, </a>
<a class="sourceLine" id="cb15-13" data-line-number="13">    <span class="dt">phred64 =</span> <span class="ot">FALSE</span>, <span class="dt">treatment =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>))</a></code></pre></div>
</div>
<div id="coverage-per-base-and-percentage-of-methylation-per-base" class="section level2">
<h2 class="hasAnchor">
<a href="#coverage-per-base-and-percentage-of-methylation-per-base" class="anchor"></a>Coverage per base and Percentage of methylation per base</h2>
<p>Histogram of percentage of methylation per cytosine for the first sample generated with the <code>methylKit</code>. Typically, the results of methylation will have a bimodal distribution, which represents the majority of bases with high or low methylation.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/hist_percent_CpG_methyl_sample1.pdf"</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="kw">getMethylationStats</span>(import.bs[[<span class="dv">1</span>]][[<span class="dv">1</span>]], <span class="dt">plot =</span> T, <span class="dt">both.strands =</span> F)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/hist_percent_CpG_methyl_sample1.png"><div align="center">
Figure 5: Percentage of methylation per base for the first sample.
</div>
<p></p>
<p>The read coverage distribution can be used as an important indication of PCR duplications bias. It is possible to observe secondary peak on the right side of the distribution if the PCR duplication bias is present.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/hist_CpG_coverage_sample1.pdf"</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="kw">getCoverageStats</span>(import.bs[[<span class="dv">1</span>]][[<span class="dv">1</span>]], <span class="dt">plot =</span> T, <span class="dt">both.strands =</span> F)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/hist_CpG_coverage_sample1.png"><div align="center">
Figure 6: Coverage base for the first sample.
</div>
<p></p>
</div>
</div>
<div id="correlation-analysis-between-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#correlation-analysis-between-samples" class="anchor"></a>Correlation analysis between samples</h1>
<p>The following computes the sample-wise <code>Spearman</code> correlation coefficients between the percentage of methylation profiles across all samples, generated with the <code>methylKit</code> package. In order to calculated the following analysis, it is necessary merge all samples, which only cytosines covered in all samples are stored.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">merged_samples &lt;-<span class="st"> </span><span class="kw">unite</span>(import.bs[[<span class="dv">1</span>]][[<span class="dv">1</span>]], <span class="dt">destrand =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>Generte table of percentage methylation for all samples.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">perc.meth &lt;-<span class="st"> </span><span class="kw">percMethylation</span>(merged_samples)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(perc.meth, <span class="st">"./results/percent_methylation.xls"</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb19-3" data-line-number="3">    <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a></code></pre></div>
<p>The correlation between samples is calculated using <code>getCorrelation</code> function. The analysis can detecting sample outliers.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/sample_correlation.pdf"</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="kw">getCorrelation</span>(merged_samples, <span class="dt">plot =</span> T, <span class="dt">method =</span> <span class="st">"spearman"</span>)</a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">get_corr &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/capture.output.html">capture.output</a></span>(<span class="kw">getCorrelation</span>(merged_samples, <span class="dt">plot =</span> F))</a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(get_corr, <span class="st">"./results/sample_correlation.xls"</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb20-6" data-line-number="6">    <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a></code></pre></div>
<img src="results/sample_correlation.png"><div align="center">
<strong>Figure 7:</strong> Sample correlation
</div>
<p></p>
<p>Also, it is possible to cluster samples hierarchically based on methylation profiles using <code>Euclidean</code> correlation distance.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/sample_clustering.pdf"</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="kw">clusterSamples</span>(merged_samples, <span class="dt">dist =</span> <span class="st">"euclidean"</span>, <span class="dt">method =</span> <span class="st">"ward"</span>, </a>
<a class="sourceLine" id="cb21-3" data-line-number="3">    <span class="dt">plot =</span> T)</a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/sample_clustering.png"><div align="center">
<strong>Figure 8:</strong> Correlation dendogram os samples.
</div>
<p></p>
<p>Principal Component Analysis (PCA) of the methylation profiles, plot shows principal component 1 and principal component 2 for each sample. Samples closer to each other in principal component space are similar in their methylation profiles.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/sample_PCA.pdf"</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">PCASamples</span>(merged_samples)</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/sample_PCA.png"><div align="center">
<strong>Figure 9:</strong> Sample PCA
</div>
<p></p>
</div>
<div id="analysis-of-differentially-methylated-regions-dmrs" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-of-differentially-methylated-regions-dmrs" class="anchor"></a>Analysis of differentially methylated regions (DMRs)</h1>
<div id="dmr-analysis-with-methylkit" class="section level2">
<h2 class="hasAnchor">
<a href="#dmr-analysis-with-methylkit" class="anchor"></a>DMR analysis with <code>MethylKit</code>
</h2>
<p>The analysis of differentially DNA methylation is calculatedby comparing methylation levels between multiple conditions, in order to show the important differences between a treatment and a control group. The following step used the <code>calculateDiffMeth</code> function from <code>MethylKit</code> package to calculate differential methylation. The samples comparisons used by this analysis are defined in the header lines of the <code>targets.txt</code> file. The calculation was based on logistic regression to calculate P-values. P-values will be adjusted to Q-values using SLIM method.</p>
<p>It is possible to select the differentially methylated regions/bases based on q-value and percent methylation difference cutoffs. The following example selects the bases that have <code>q-value &lt; 0.01</code> and percent methylation difference larger than 25%.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(methylKit)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"R_function/run_methylKit.R"</span>)</a>
<a class="sourceLine" id="cb23-3" data-line-number="3">targetspath &lt;-<span class="st"> "targets.txt"</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5">cmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span>(<span class="dt">file =</span> targetspath, <span class="dt">format =</span> <span class="st">"matrix"</span>, <span class="dt">delim =</span> <span class="st">"-"</span>)</a>
<a class="sourceLine" id="cb23-6" data-line-number="6">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/bismark_methyl_extractor.param"</span>, </a>
<a class="sourceLine" id="cb23-7" data-line-number="7">    <span class="dt">mytargets =</span> <span class="st">"targets_bismark.txt"</span>)</a>
<a class="sourceLine" id="cb23-8" data-line-number="8">dmr.DF &lt;-<span class="st"> </span><span class="kw">run_methylKit</span>(targets, args, <span class="dt">cmp =</span> cmp[[<span class="dv">1</span>]], <span class="dt">assembly =</span> <span class="st">"TAIR10"</span>, </a>
<a class="sourceLine" id="cb23-9" data-line-number="9">    <span class="dt">filter =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">qvalue =</span> <span class="fl">0.01</span>, <span class="dt">difference =</span> <span class="dv">25</span>), <span class="dt">plot =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<img src="results/DMR_methyl.png"><div align="center">
<strong>Figure 10:</strong> Differentially methylated regions (DMRs)
</div>
<p></p>
</div>
<div id="dmr-analysis-with-dss" class="section level2">
<h2 class="hasAnchor">
<a href="#dmr-analysis-with-dss" class="anchor"></a>DMR analysis with <code>DSS</code>
</h2>
<p>Perform statistical test for DML by calling DMLtest function. This function basically performs following steps: (1) estimate mean methylation levels for all CpG site; (2) estimate dispersions at each CpG sites; (3) conduct Wald test. For the first step, there’s an option for smoothing or not. Because the methylation levels show strong spatial correlations, smoothing can help obtain better estimates of mean methylation when the CpG sites are dense in the data (such as from the whole-genome BS-seq). However for data with sparse CpG, such as from RRBS or hydroxyl-methylation, smoothing is not recommended.</p>
<p>User has the option to smooth the methylation levels or not. For WGBS data, smoothing is recommended so that information from nearby CpG sites can be combined to improve the estimation of methylation levels. A simple moving average algorithm is implemented for smoothing. In RRBS since the CpG coverage is sparse, smoothing might not alter the results much. If smoothing is requested, smoothing span is an important parameter which has non-trivial impact on DMR calling. We use 500 bp as default, and think that it performs well in real data tests.</p>
<p>With the test results, one can call DML by using callDML function. The results DMLs are sorted by the significance.</p>
<p>DMR detection is also Based on the DML test results, by calling callDMR function. Regions with many statistically significant CpG sites are identified as DMRs. Some restrictions are provided by users, including the minimum length, minimum number of CpG sites, percentage of CpG site being significant in the region, etc. There are some post hoc procedures to merge nearby DMRs into longer ones.</p>
<p>Note that the distribution of test statistics (and p-values) depends on the differences in methylation levels and biological variations, as well as technical factors such as coverage depth. It is very difficulty to select a natural and rigorous threshold for defining DMRs. We recommend users try different thresholds in order to obtain satisfactory results. 5. The DMRs can be visualized using showOneDMR function, This function provides more information than the plotRegion function in bsseq. It plots the methylation percentages as well as the coverage depths at each CpG sites, instead of just the smoothed curve. So the coverage depth information will be available in the figure. To use the function, do</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(DSS)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"R_function/run_dds.R"</span>)</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">targetspath &lt;-<span class="st"> "targets.txt"</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb24-5" data-line-number="5">cmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span>(<span class="dt">file =</span> targetspath, <span class="dt">format =</span> <span class="st">"matrix"</span>, <span class="dt">delim =</span> <span class="st">"-"</span>)</a>
<a class="sourceLine" id="cb24-6" data-line-number="6">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/bismark_methyl_extractor.param"</span>, </a>
<a class="sourceLine" id="cb24-7" data-line-number="7">    <span class="dt">mytargets =</span> <span class="st">"targets_bismark.txt"</span>)</a>
<a class="sourceLine" id="cb24-8" data-line-number="8">dss.DF &lt;-<span class="st"> </span><span class="kw">run_dss</span>(targets, args, <span class="dt">cmp =</span> cmp[[<span class="dv">1</span>]], <span class="dt">smoothing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb24-9" data-line-number="9"></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">## Calling DML</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11">dmls &lt;-<span class="st"> </span><span class="kw">callDML</span>(dss.DF[[<span class="dv">1</span>]], <span class="dt">delta =</span> <span class="fl">0.1</span>, <span class="dt">p.threshold =</span> <span class="fl">0.001</span>)</a>
<a class="sourceLine" id="cb24-12" data-line-number="12"></a>
<a class="sourceLine" id="cb24-13" data-line-number="13"><span class="co">## Calling DMR</span></a>
<a class="sourceLine" id="cb24-14" data-line-number="14">dss.obj &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span>(<span class="st">"./results/dss.input_C-S.rds"</span>)</a>
<a class="sourceLine" id="cb24-15" data-line-number="15">dmrs &lt;-<span class="st"> </span><span class="kw">callDMR</span>(dss.DF[[<span class="dv">1</span>]], <span class="dt">delta =</span> <span class="fl">0.1</span>, <span class="dt">p.threshold =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb24-16" data-line-number="16"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/plotRegion.DMR.pdf"</span>)</a>
<a class="sourceLine" id="cb24-17" data-line-number="17"><span class="kw">showOneDMR</span>(dmrs[<span class="dv">1</span>, ], dss.obj)</a>
<a class="sourceLine" id="cb24-18" data-line-number="18"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<p><img src="results/sample_PCA.png"></p>
<div align="center">
<strong>Figure 10:</strong> Methylation percentages and the coverage depths at each CpG sites.
</div>
<p></p>
</div>
<div id="dmr-analysis-with-biseq" class="section level2">
<h2 class="hasAnchor">
<a href="#dmr-analysis-with-biseq" class="anchor"></a>DMR analysis with <code>Biseq</code>
</h2>
<div id="import-of-bismarks-methylation-output-files-with-biseq-package-biseq-" class="section level3">
<h3 class="hasAnchor">
<a href="#import-of-bismarks-methylation-output-files-with-biseq-package-biseq-" class="anchor"></a>Import of Bismark’s methylation output files with <code>BiSeq</code> package <span class="citation">(Hebestreit and Klein 2015)</span>.</h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BiSeq)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>(), <span class="st">"/results"</span>, <span class="dt">sep =</span> <span class="st">""</span>)</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">report &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()</a>
<a class="sourceLine" id="cb25-4" data-line-number="4">file.list &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(path, <span class="st">"*.extractor"</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb25-5" data-line-number="5">report &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()</a>
<a class="sourceLine" id="cb25-6" data-line-number="6"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> file.list)) {</a>
<a class="sourceLine" id="cb25-7" data-line-number="7">    file.list1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(file.list[i]), <span class="st">".cov"</span>, </a>
<a class="sourceLine" id="cb25-8" data-line-number="8">        <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb25-9" data-line-number="9">    report[<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(report) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>] &lt;-<span class="st"> </span>file.list1</a>
<a class="sourceLine" id="cb25-10" data-line-number="10">}</a>
<a class="sourceLine" id="cb25-11" data-line-number="11">colData &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(<span class="dt">group =</span> <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Control"</span>, <span class="st">"Control"</span>, <span class="st">"Control"</span>, </a>
<a class="sourceLine" id="cb25-12" data-line-number="12">    <span class="st">"Stress"</span>, <span class="st">"Stress"</span>, <span class="st">"Stress"</span>)), <span class="dt">row.names =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>, <span class="st">"C2"</span>, </a>
<a class="sourceLine" id="cb25-13" data-line-number="13">    <span class="st">"C3"</span>, <span class="st">"S1"</span>, <span class="st">"S2"</span>, <span class="st">"S3"</span>))</a>
<a class="sourceLine" id="cb25-14" data-line-number="14">wgbs &lt;-<span class="st"> </span><span class="kw">readBismark</span>(<span class="dt">files =</span> <span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(report), colData)</a></code></pre></div>
<p>Number of CpG sites that were covered per sample together with the median of the coverage of these CpG sites and the representation of the coverage distributions per sample.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw">covStatistics</span>(wgbs)</a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/CoverangePerSample_biseq.pdf"</span>)</a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="kw">covBoxplots</span>(wgbs, <span class="dt">col =</span> <span class="st">"cornflowerblue"</span>, <span class="dt">las =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<p>The following detect differentially methylation regions within two groups os samples. First, we detect CpG clusters</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">wgbs.small &lt;-<span class="st"> </span>wgbs[<span class="dv">1</span><span class="op">:</span><span class="dv">10000</span>, ]</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">wgbs.clust.unlim &lt;-<span class="st"> </span><span class="kw">clusterSites</span>(<span class="dt">object =</span> wgbs.small, <span class="dt">perc.samples =</span> <span class="dv">4</span><span class="op">/</span><span class="dv">6</span>, </a>
<a class="sourceLine" id="cb27-3" data-line-number="3">    <span class="dt">min.sites =</span> <span class="dv">20</span>, <span class="dt">max.dist =</span> <span class="dv">100</span>, <span class="dt">minCov =</span> <span class="dv">6</span>)</a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">rowRanges</span>(wgbs.clust.unlim))</a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="kw">clusterSitesToGR</span>(wgbs.clust.unlim)</a></code></pre></div>
<p>Next, smooth methylation data within CpG clusters, in order to reduce bias due to high coverages. The effect of the smoothing step can be plot.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">ind.cov &lt;-<span class="st"> </span><span class="kw">totalReads</span>(wgbs.clust.unlim) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2">quant &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="kw">totalReads</span>(wgbs.clust.unlim)[ind.cov], <span class="fl">0.9</span>)</a>
<a class="sourceLine" id="cb28-3" data-line-number="3">quant</a>
<a class="sourceLine" id="cb28-4" data-line-number="4">wgbs.clust.lim &lt;-<span class="st"> </span><span class="kw">limitCov</span>(wgbs.clust.unlim, <span class="dt">maxCov =</span> quant)</a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="kw">covBoxplots</span>(wgbs.clust.lim, <span class="dt">col =</span> <span class="st">"cornflowerblue"</span>, <span class="dt">las =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb28-6" data-line-number="6">predictedMeth &lt;-<span class="st"> </span><span class="kw">predictMeth</span>(<span class="dt">object =</span> wgbs.clust.lim)</a>
<a class="sourceLine" id="cb28-7" data-line-number="7">predictedMeth</a>
<a class="sourceLine" id="cb28-8" data-line-number="8">region &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="dt">seqnames =</span> <span class="st">"5"</span>, <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="dt">start =</span> <span class="dv">100</span>, </a>
<a class="sourceLine" id="cb28-9" data-line-number="9">    <span class="dt">end =</span> <span class="dv">830</span>))</a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/smoothed_biseq.pdf"</span>)</a>
<a class="sourceLine" id="cb28-11" data-line-number="11"><span class="kw">plotMeth</span>(<span class="dt">object.raw =</span> wgbs[, <span class="dv">2</span>], <span class="dt">object.rel =</span> predictedMeth[, </a>
<a class="sourceLine" id="cb28-12" data-line-number="12">    <span class="dv">2</span>], <span class="dt">region =</span> region, <span class="dt">lwd.lines =</span> <span class="dv">2</span>, <span class="dt">col.points =</span> <span class="st">"blue"</span>, </a>
<a class="sourceLine" id="cb28-13" data-line-number="13">    <span class="dt">cex =</span> <span class="fl">1.5</span>)</a>
<a class="sourceLine" id="cb28-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/smoothed_biseq.png"><div align="center">
<strong>Figure 11:</strong> Smooth methylation levels.
</div>
<p></p>
<p>The follow show the differential methylation between the comparison for some CpG sites. Test CpG clusters for differential methylation and control weighted FDR on cluster.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">control &lt;-<span class="st"> </span>predictedMeth[, <span class="kw">colData</span>(predictedMeth)<span class="op">$</span>group <span class="op">==</span><span class="st"> "Control"</span>]</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">stress &lt;-<span class="st"> </span>predictedMeth[, <span class="kw">colData</span>(predictedMeth)<span class="op">$</span>group <span class="op">==</span><span class="st"> "Stress"</span>]</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">mean.control &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="kw">methLevel</span>(control))</a>
<a class="sourceLine" id="cb29-4" data-line-number="4">mean.stress &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="kw">methLevel</span>(stress))</a>
<a class="sourceLine" id="cb29-5" data-line-number="5">betaResults &lt;-<span class="st"> </span><span class="kw">betaRegression</span>(<span class="dt">formula =</span> <span class="op">~</span>group, <span class="dt">link =</span> <span class="st">"probit"</span>, </a>
<a class="sourceLine" id="cb29-6" data-line-number="6">    <span class="dt">object =</span> predictedMeth, <span class="dt">type =</span> <span class="st">"BR"</span>)</a>
<a class="sourceLine" id="cb29-7" data-line-number="7"></a>
<a class="sourceLine" id="cb29-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(betaResults)</a>
<a class="sourceLine" id="cb29-9" data-line-number="9">vario &lt;-<span class="st"> </span><span class="kw">makeVariogram</span>(betaResults)</a>
<a class="sourceLine" id="cb29-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/smoothed_levels_biseq.pdf"</span>)</a>
<a class="sourceLine" id="cb29-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(mean.stress, mean.control, <span class="dt">col =</span> <span class="st">"blue"</span>, <span class="dt">xlab =</span> <span class="st">"Methylation in Controls"</span>, </a>
<a class="sourceLine" id="cb29-12" data-line-number="12">    <span class="dt">ylab =</span> <span class="st">"Methylation in Stress"</span>)</a>
<a class="sourceLine" id="cb29-13" data-line-number="13"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/smoothed_levels_biseq.png"><div align="center">
<strong>Figure 12:</strong> Smooth methylation levels between control and stress samples..
</div>
<p></p>
<p>Detect CpG clusters containing at least one differentially methylated location.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">predictedMethNull &lt;-<span class="st"> </span>predictedMeth[<span class="dv">1</span><span class="op">:</span><span class="dv">800</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw">colData</span>(predictedMethNull)<span class="op">$</span>group.null &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb30-3" data-line-number="3">betaResultsNull &lt;-<span class="st"> </span><span class="kw">betaRegression</span>(<span class="dt">formula =</span> <span class="op">~</span>group.null, <span class="dt">link =</span> <span class="st">"probit"</span>, </a>
<a class="sourceLine" id="cb30-4" data-line-number="4">    <span class="dt">object =</span> predictedMethNull, <span class="dt">type =</span> <span class="st">"BR"</span>)</a>
<a class="sourceLine" id="cb30-5" data-line-number="5">vario &lt;-<span class="st"> </span><span class="kw">makeVariogram</span>(betaResultsNull)</a>
<a class="sourceLine" id="cb30-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/estimated_variogram_biseq.pdf"</span>)</a>
<a class="sourceLine" id="cb30-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(vario<span class="op">$</span>variogram<span class="op">$</span>v)</a>
<a class="sourceLine" id="cb30-8" data-line-number="8">vario.sm &lt;-<span class="st"> </span><span class="kw">smoothVariogram</span>(vario, <span class="dt">sill =</span> <span class="fl">0.9</span>)</a>
<a class="sourceLine" id="cb30-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(vario.sm<span class="op">$</span>variogram[, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"h"</span>, <span class="st">"v.sm"</span>)], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">lwd =</span> <span class="fl">1.5</span>)</a>
<a class="sourceLine" id="cb30-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span>()</a>
<a class="sourceLine" id="cb30-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<p>Next, we trimed the rejected CpG clusters that is to remove the not differentially methylated CpG sites at q1 what can be interpreted as the location-wise FDR. Also, it is possible define the boundaries of DMRs as rejected CpG sites within which rejected CpG sites solely are located.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">vario.aux &lt;-<span class="st"> </span><span class="kw">makeVariogram</span>(betaResults, <span class="dt">make.variogram =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb31-2" data-line-number="2">vario.sm<span class="op">$</span>pValsList &lt;-<span class="st"> </span>vario.aux<span class="op">$</span>pValsList</a>
<a class="sourceLine" id="cb31-3" data-line-number="3">locCor &lt;-<span class="st"> </span><span class="kw">estLocCor</span>(vario.sm)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4">clusters.rej &lt;-<span class="st"> </span><span class="kw">testClusters</span>(locCor, <span class="dt">FDR.cluster =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb31-5" data-line-number="5">clusters.trimmed &lt;-<span class="st"> </span><span class="kw">trimClusters</span>(clusters.rej, <span class="dt">FDR.loc =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb31-6" data-line-number="6">DMRs &lt;-<span class="st"> </span><span class="kw">findDMRs</span>(clusters.trimmed, <span class="dt">max.dist =</span> <span class="dv">100</span>, <span class="dt">diff.dir =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb31-7" data-line-number="7">DMRs<span class="fl">.2</span> &lt;-<span class="st"> </span><span class="kw">compareTwoSamples</span>(<span class="dt">object =</span> predictedMeth, <span class="dt">sample1 =</span> <span class="st">"C1"</span>, </a>
<a class="sourceLine" id="cb31-8" data-line-number="8">    <span class="dt">sample2 =</span> <span class="st">"S1"</span>, <span class="dt">minDiff =</span> <span class="fl">0.3</span>, <span class="dt">max.dist =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb31-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="kw">overlapsAny</span>(DMRs<span class="fl">.2</span>, DMRs))</a></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">rowCols &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"magenta"</span>, <span class="st">"blue"</span>)[<span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="kw">colData</span>(predictedMeth)<span class="op">$</span>group)]</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="kw">plotMethMap</span>(predictedMeth, <span class="dt">region =</span> DMRs[<span class="dv">3</span>], <span class="dt">groups =</span> <span class="kw">colData</span>(predictedMeth)<span class="op">$</span>group, </a>
<a class="sourceLine" id="cb32-3" data-line-number="3">    <span class="dt">intervals =</span> <span class="ot">FALSE</span>, <span class="dt">zlim =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">RowSideColors =</span> rowCols, </a>
<a class="sourceLine" id="cb32-4" data-line-number="4">    <span class="dt">labCol =</span> <span class="st">""</span>, <span class="dt">margins =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>, <span class="dv">6</span>))</a></code></pre></div>
</div>
</div>
</div>
<div id="annotation-of-differentially-methylated-regions-dmrs" class="section level1">
<h1 class="hasAnchor">
<a href="#annotation-of-differentially-methylated-regions-dmrs" class="anchor"></a>Annotation of differentially methylated regions (DMRs)</h1>
<div id="annotation-with-genomation-package" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-with-genomation-package" class="anchor"></a>Annotation with <code>genomation</code> package</h2>
<p>differentially methylated regions/bas based on gene annotation using genomation package</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GenomicFeatures"</span>)</a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BiocParallel)</a>
<a class="sourceLine" id="cb33-3" data-line-number="3">txdb &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html">makeTxDbFromGFF</a></span>(<span class="dt">file =</span> <span class="st">"data/TAIR10_GFF3_genes.gff"</span>, </a>
<a class="sourceLine" id="cb33-4" data-line-number="4">    <span class="dt">format =</span> <span class="st">"gff"</span>, <span class="dt">dataSource =</span> <span class="st">"TAIR"</span>, <span class="dt">organism =</span> <span class="st">"Arabidopsis thaliana"</span>)</a>
<a class="sourceLine" id="cb33-5" data-line-number="5"><span class="kw">saveDb</span>(txdb, <span class="dt">file =</span> <span class="st">"./data/tair10.sqlite"</span>)</a>
<a class="sourceLine" id="cb33-6" data-line-number="6">txdb &lt;-<span class="st"> </span><span class="kw">loadDb</span>(<span class="st">"./data/tair10.sqlite"</span>)</a>
<a class="sourceLine" id="cb33-7" data-line-number="7">feat &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/genFeatures.html">genFeatures</a></span>(txdb, <span class="dt">featuretype =</span> <span class="st">"all"</span>, <span class="dt">reduce_ranges =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb33-8" data-line-number="8">    <span class="dt">upstream =</span> <span class="dv">1000</span>, <span class="dt">downstream =</span> <span class="dv">0</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb33-9" data-line-number="9">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/bismark_aligner.param"</span>, <span class="dt">mytargets =</span> <span class="st">"targets_trim.txt"</span>)</a>
<a class="sourceLine" id="cb33-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb33-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(grid)</a>
<a class="sourceLine" id="cb33-12" data-line-number="12"></a>
<a class="sourceLine" id="cb33-13" data-line-number="13">path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>(), <span class="st">"/results"</span>, <span class="dt">sep =</span> <span class="st">""</span>)</a>
<a class="sourceLine" id="cb33-14" data-line-number="14">report &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()</a>
<a class="sourceLine" id="cb33-15" data-line-number="15">file.list &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(path, <span class="st">"*.out"</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb33-16" data-line-number="16">report &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>()</a>
<a class="sourceLine" id="cb33-17" data-line-number="17"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> file.list)) {</a>
<a class="sourceLine" id="cb33-18" data-line-number="18">    file.list1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(file.list[i]), <span class="st">"*.bam"</span>, </a>
<a class="sourceLine" id="cb33-19" data-line-number="19">        <span class="dt">full.names =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb33-20" data-line-number="20">    report[<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(report) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>] &lt;-<span class="st"> </span>file.list1</a>
<a class="sourceLine" id="cb33-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb33-22" data-line-number="22"></a>
<a class="sourceLine" id="cb33-23" data-line-number="23">fc &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/featuretypeCounts.html">featuretypeCounts</a></span>(<span class="dt">bfl =</span> <span class="kw">BamFileList</span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(report), </a>
<a class="sourceLine" id="cb33-24" data-line-number="24">    <span class="dt">yieldSize =</span> <span class="dv">50000</span>), <span class="dt">grl =</span> feat, <span class="dt">singleEnd =</span> <span class="ot">TRUE</span>, <span class="dt">readlength =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb33-25" data-line-number="25">    <span class="dt">type =</span> <span class="st">"data.frame"</span>)</a>
<a class="sourceLine" id="cb33-26" data-line-number="26"></a>
<a class="sourceLine" id="cb33-27" data-line-number="27">p &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/plotfeaturetypeCounts.html">plotfeaturetypeCounts</a></span>(<span class="dt">x =</span> fc, <span class="dt">graphicsfile =</span> <span class="st">"results/featureCounts.png"</span>, </a>
<a class="sourceLine" id="cb33-28" data-line-number="28">    <span class="dt">graphicsformat =</span> <span class="st">"png"</span>, <span class="dt">scales =</span> <span class="st">"fixed"</span>, <span class="dt">anyreadlength =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb33-29" data-line-number="29">    <span class="dt">scale_length_val =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb33-30" data-line-number="30"></a>
<a class="sourceLine" id="cb33-31" data-line-number="31"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(genomation)</a></code></pre></div>
</div>
<div id="annotation-with-chippeakanno-package" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-with-chippeakanno-package" class="anchor"></a>Annotation with <code>ChIPpeakAnno</code> package</h2>
<p>The following annotates the DMR with genomic context information using the <code>ChIPpeakAnno</code> package <span class="citation">(Zhu et al. 2010)</span>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ChIPpeakAnno)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicFeatures)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/systemArgs.html">systemArgs</a></span>(<span class="dt">sysma =</span> <span class="st">"param/annotate_peaks.param"</span>, <span class="dt">mytargets =</span> <span class="st">"targets_macs.txt"</span>)</a>
<a class="sourceLine" id="cb34-4" data-line-number="4"><span class="co"># txdb &lt;- loadDb('./data/tair10.sqlite')</span></a>
<a class="sourceLine" id="cb34-5" data-line-number="5">txdb &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html">makeTxDbFromGFF</a></span>(<span class="dt">file =</span> <span class="st">"data/TAIR10_GFF3_genes.gff"</span>, </a>
<a class="sourceLine" id="cb34-6" data-line-number="6">    <span class="dt">format =</span> <span class="st">"gff"</span>, <span class="dt">dataSource =</span> <span class="st">"TAIR"</span>, <span class="dt">organism =</span> <span class="st">"Arabidopsis thaliana"</span>)</a>
<a class="sourceLine" id="cb34-7" data-line-number="7">ge &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span>(txdb, <span class="dt">columns =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"tx_name"</span>, <span class="st">"gene_id"</span>, <span class="st">"tx_type"</span>))</a>
<a class="sourceLine" id="cb34-8" data-line-number="8"></a>
<a class="sourceLine" id="cb34-9" data-line-number="9"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> args)) {</a>
<a class="sourceLine" id="cb34-10" data-line-number="10">    peaksGR &lt;-<span class="st"> </span><span class="kw">as</span>(merged_samples, <span class="st">"GRanges"</span>)</a>
<a class="sourceLine" id="cb34-11" data-line-number="11">    annotatedPeak &lt;-<span class="st"> </span><span class="kw">annotatePeakInBatch</span>(peaksGR, <span class="dt">AnnotationData =</span> <span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span>(txdb))</a>
<a class="sourceLine" id="cb34-12" data-line-number="12">    df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(annotatedPeak), <span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw">values</span>(ge[<span class="kw">values</span>(annotatedPeak)<span class="op">$</span>feature, </a>
<a class="sourceLine" id="cb34-13" data-line-number="13">        ])))</a>
<a class="sourceLine" id="cb34-14" data-line-number="14">    <span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(df, <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">outpaths</a></span>(args[i]), <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb34-15" data-line-number="15">        <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a>
<a class="sourceLine" id="cb34-16" data-line-number="16">}</a>
<a class="sourceLine" id="cb34-17" data-line-number="17"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_peakanno.txt"</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R Under development (unstable) (2020-02-05 r77773)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 19.04
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so
## LAPACK: /usr/local/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices
## [6] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] ape_5.3                     ggplot2_3.2.1              
##  [3] batchtools_0.9.12           systemPipeR_1.21.2         
##  [5] ShortRead_1.45.3            GenomicAlignments_1.23.1   
##  [7] SummarizedExperiment_1.17.2 DelayedArray_0.13.4        
##  [9] matrixStats_0.55.0          Biobase_2.47.2             
## [11] BiocParallel_1.21.2         Rsamtools_2.3.4            
## [13] Biostrings_2.55.4           XVector_0.27.0             
## [15] GenomicRanges_1.39.2        GenomeInfoDb_1.23.13       
## [17] IRanges_2.21.3              S4Vectors_0.25.12          
## [19] BiocGenerics_0.33.0         BiocStyle_2.15.6           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1         rjson_0.2.20            
##   [3] hwriter_1.3.2            rprojroot_1.3-2         
##   [5] fs_1.3.1                 rstudioapi_0.11         
##   [7] bit64_0.9-7              AnnotationDbi_1.49.1    
##   [9] codetools_0.2-16         splines_4.0.0           
##  [11] knitr_1.28               annotate_1.65.1         
##  [13] GO.db_3.10.0             dbplyr_1.4.2            
##  [15] png_0.1-7                pheatmap_1.0.12         
##  [17] graph_1.65.1             BiocManager_1.30.10     
##  [19] compiler_4.0.0           httr_1.4.1              
##  [21] GOstats_2.53.0           backports_1.1.5         
##  [23] assertthat_0.2.1         Matrix_1.2-18           
##  [25] lazyeval_0.2.2           limma_3.43.4            
##  [27] formatR_1.7              htmltools_0.4.0         
##  [29] prettyunits_1.1.1        tools_4.0.0             
##  [31] gtable_0.3.0             glue_1.3.1              
##  [33] GenomeInfoDbData_1.2.2   Category_2.53.1         
##  [35] dplyr_0.8.4              rappdirs_0.3.1          
##  [37] Rcpp_1.0.3               pkgdown_1.4.1           
##  [39] vctrs_0.2.3              nlme_3.1-144            
##  [41] rtracklayer_1.47.0       xfun_0.12               
##  [43] stringr_1.4.0            lifecycle_0.1.0         
##  [45] XML_3.99-0.3             edgeR_3.29.0            
##  [47] zlibbioc_1.33.1          MASS_7.3-51.5           
##  [49] scales_1.1.0             BSgenome_1.55.3         
##  [51] VariantAnnotation_1.33.0 hms_0.5.3               
##  [53] RBGL_1.63.1              RColorBrewer_1.1-2      
##  [55] yaml_2.2.1               curl_4.3                
##  [57] memoise_1.1.0            biomaRt_2.43.3          
##  [59] latticeExtra_0.6-29      stringi_1.4.6           
##  [61] RSQLite_2.2.0            genefilter_1.69.0       
##  [63] desc_1.2.0               checkmate_2.0.0         
##  [65] GenomicFeatures_1.39.5   rlang_0.4.4             
##  [67] pkgconfig_2.0.3          bitops_1.0-6            
##  [69] evaluate_0.14            lattice_0.20-38         
##  [71] purrr_0.3.3              bit_1.1-15.2            
##  [73] tidyselect_1.0.0         GSEABase_1.49.0         
##  [75] AnnotationForge_1.29.2   magrittr_1.5            
##  [77] bookdown_0.17            R6_2.4.1                
##  [79] base64url_1.4            DBI_1.1.0               
##  [81] withr_2.1.2              pillar_1.4.3            
##  [83] survival_3.1-8           RCurl_1.98-1.1          
##  [85] tibble_2.1.3             crayon_1.3.4            
##  [87] BiocFileCache_1.11.4     rmarkdown_2.1           
##  [89] jpeg_0.1-8.1             progress_1.2.2          
##  [91] locfit_1.5-9.1           grid_4.0.0              
##  [93] data.table_1.12.8        blob_1.2.1              
##  [95] Rgraphviz_2.31.0         digest_0.6.24           
##  [97] xtable_1.8-4             brew_1.0-6              
##  [99] openssl_1.4.1            munsell_0.5.0           
## [101] askpass_1.1</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Institutes of Health (NIH) and the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Ganguly2017-ur">
<p>Ganguly, Diep R, Peter A Crisp, Steven R Eichten, and Barry J Pogson. 2017. “The Arabidopsis DNA Methylome Is Stable Under Transgenerational Drought Stress.” <em>Plant Physiol.</em> 175 (4): 1893–1912.</p>
</div>
<div id="ref-Girke2014-oy">
<p>Girke, Thomas. 2014. “systemPipeR: NGS Workflow and Report Generation Environment.” UC Riverside. <a href="https://github.com/tgirke/systemPipeR">https://github.com/tgirke/systemPipeR</a>.</p>
</div>
<div id="ref-biseq">
<p>Hebestreit, Katja, and Hans-Ulrich Klein. 2015. <em>BiSeq: Processing and Analyzing Bisulfite Sequencing Data</em>.</p>
</div>
<div id="ref-Krueger2011-uu">
<p>Krueger, Felix, and Simon R Andrews. 2011. “Bismark: A Flexible Aligner and Methylation Caller for Bisulfite-Seq Applications.” <em>Bioinformatics</em> 27 (11): 1571–2.</p>
</div>
<div id="ref-Langmead2012-bs">
<p>Langmead, Ben, and Steven L Salzberg. 2012. “Fast Gapped-Read Alignment with Bowtie 2.” <em>Nat. Methods</em> 9 (4). Nature Publishing Group: 357–59. <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.</p>
</div>
<div id="ref-Zhu2010-zo">
<p>Zhu, Lihua J, Claude Gazin, Nathan D Lawson, Hervé Pagès, Simon M Lin, David S Lapointe, and Michael R Green. 2010. “ChIPpeakAnno: A Bioconductor Package to Annotate ChIP-seq and ChIP-chip Data.” <em>BMC Bioinformatics</em> 11: 237. <a href="https://doi.org/10.1186/1471-2105-11-237">https://doi.org/10.1186/1471-2105-11-237</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#sample-definitions-and-environment-settings">Sample definitions and environment settings</a><ul class="nav nav-pills nav-stacked">
<li><a href="#environment-settings-and-input-data">Environment settings and input data</a></li>
      <li><a href="#required-packages-and-resources">Required packages and resources</a></li>
      <li><a href="#experiment-definition-provided-by-targets-file">Experiment definition provided by <code>targets</code> file</a></li>
      </ul>
</li>
      <li>
<a href="#read-preprocessing">Read preprocessing</a><ul class="nav nav-pills nav-stacked">
<li><a href="#read-quality-filtering-and-trimming">Read quality filtering and trimming</a></li>
      <li><a href="#fastq-quality-report">FASTQ quality report</a></li>
      </ul>
</li>
      <li>
<a href="#alignments">Alignments</a><ul class="nav nav-pills nav-stacked">
<li><a href="#read-mapping-with-bowtiebismark">Read mapping with <code>Bowtie/Bismark</code></a></li>
      <li><a href="#read-and-plot-the-alignment-stats">Read and plot the alignment stats</a></li>
      <li><a href="#methylation-calling">Methylation Calling</a></li>
      </ul>
</li>
      <li>
<a href="#descriptive-statistics-on-dna-methylation-profiles">Descriptive statistics on DNA methylation profiles</a><ul class="nav nav-pills nav-stacked">
<li><a href="#import-of-bismark-methylation-output-files">Import of Bismark methylation output files</a></li>
      <li><a href="#coverage-per-base-and-percentage-of-methylation-per-base">Coverage per base and Percentage of methylation per base</a></li>
      </ul>
</li>
      <li><a href="#correlation-analysis-between-samples">Correlation analysis between samples</a></li>
      <li>
<a href="#analysis-of-differentially-methylated-regions-dmrs">Analysis of differentially methylated regions (DMRs)</a><ul class="nav nav-pills nav-stacked">
<li><a href="#dmr-analysis-with-methylkit">DMR analysis with <code>MethylKit</code></a></li>
      <li><a href="#dmr-analysis-with-dss">DMR analysis with <code>DSS</code></a></li>
      <li><a href="#dmr-analysis-with-biseq">DMR analysis with <code>Biseq</code></a></li>
      </ul>
</li>
      <li>
<a href="#annotation-of-differentially-methylated-regions-dmrs">Annotation of differentially methylated regions (DMRs)</a><ul class="nav nav-pills nav-stacked">
<li><a href="#annotation-with-genomation-package">Annotation with <code>genomation</code> package</a></li>
      <li><a href="#annotation-with-chippeakanno-package">Annotation with <code>ChIPpeakAnno</code> package</a></li>
      </ul>
</li>
      <li><a href="#version-information">Version Information</a></li>
      <li><a href="#funding">Funding</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '',
    indexName: '',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
